src/modules/jobs/models/tech-load.model.ts

```diff
@@ -0,0 +1,399 @@
+      else {
+        availableSpeeds = [10, 15, 25];
+        if (transportType.match(/BP/i)) {
+          availableSpeeds.push(50);
+          availableSpeeds.push(100);
+        }
+      }
+      services.internetSpeed = this.randOf<number>(availableSpeeds);
+    }
+
+    if (this.nextRand() < 0.05) { // Add pots
+      services.potsNumber = this.randDigits(10);
+    }
+    else if (this.nextRand() < 0.2) { // Add voip
+      services.voipNumbers = [this.randDigits(10)];
+      if (this.nextRand() < 0.01) {
+        services.voipNumbers.push(this.randDigits(10));
+      }
+    }
+
+    return services;
+  }
+
+  /**
+   * @TODO make sure single-line location string works
+   * @description - Randomly draws location data from a mini database of locations. Is just all 50 state capitals and their lat/long coordinates.
+   * @param {string} streetAddress - The customer's street address, to be included in the city location data.
+   * @returns {Location} - The generated city data, which includes the street address, city and state names, and lat/long coordinates.
+   */
+  private generateCityLocation(streetAddress: string): Location {
+    const locations = `AL,Montgomery,32.377716,-86.300568;AK,Juneau,58.301598,-134.420212;AZ,Phoenix,33.448143,-112.096962;AR,Little Rock,34.746613,-92.288986;CA,Sacramento,38.576668,-121.493629;CO,Denver,39.739227,-104.984856;CT,Hartford,41.764046,-72.682198;DE,Dover,39.157307,-75.519722;HI,Honolulu,21.307442,-157.857376;FL,Tallahassee,30.438118,-84.281296;GA,Atlanta,33.749027,-84.388229;ID,Boise,43.617775,-116.199722;IL,Springfield,39.798363,-89.654961;IN,Indianapolis,39.768623,-86.162643;IA,Des Moines,41.591087,-93.603729;KS,Topeka,39.048191,-95.677956;KY,Frankfort,38.186722,-84.875374;LA,Baton Rouge,30.457069,-91.187393;ME,Augusta,44.307167,-69.781693;MD,Annapolis,38.978764,-76.490936;MA,Boston,42.358162,-71.063698;MI,Lansing,42.733635,-84.555328;MN,St. Paul,44.955097,-93.102211;MS,Jackson,32.303848,-90.182106;MO,Jefferson City,38.579201,-92.172935;MT,Helena,46.585709,-112.018417;NE,Lincoln,40.808075,-96.699654;NV,Carson City,39.163914,-119.766121;NH,Concord,43.206898,-71.537994;NJ,Trenton,40.220596,-74.769913;NM,Santa Fe,35.68224,-105.939728;NC,Raleigh,35.78043,-78.639099;ND,Bismarck,46.82085,-100.783318;NY,Albany,42.652843,-73.757874;OH,Columbus,39.961346,-82.999069;OK,Oklahoma City,35.492207,-97.503342;OR,Salem,44.938461,-123.030403;PA,Harrisburg,40.264378,-76.883598;RI,Providence,41.830914,-71.414963;SC,Columbia,34.000343,-81.033211;SD,Pierre,44.367031,-100.346405;TN,Nashville,36.16581,-86.784241;TX,Austin,30.27467,-97.740349;UT,Salt Lake City,40.777477,-111.888237;VT,Montpelier,44.262436,-72.580536;VA,Richmond,37.538857,-77.43364;WA,Olympia,47.035805,-122.905014;WV,Charleston,38.336246,-81.612328;WI,Madison,43.074684,-89.384445;WY,Cheyenne,41.140259,-104.820236`
+      .split(';').map(function(el: string): Location { // kinda gross, but this basically converts the above string into an array of City objects
```

Comment:
Any reason using `function() {}` rather than arrow function?

Reply:
I think I could't figure out where to put the return type lol

Reply:
I've done this before hahaha

Reply:
I figured it out.